drivers/hdlc_rcp_if/Kconfig.nxp

```diff
@@ -0,0 +1,9 @@
+# Openthead HDLC RCP communication Interface configuration options
+# Copyright 2024 NXP
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+
+menuconfig OPENTHREAD_HDLC_RCP
```

Comment:
I think this shoudl be defined as a generic config in `net/l2/openthread/Kconfig`. Generic OpenThread code (which ends up in `modules/openthread` should not depend on vendor-specific options.

Reply:
Hi @rlubos,
I am correcting the PR regarding your comments.
I don't understand your remark here because the Kconfig.nxp is dedicated for the NXP vendor.
If a new XXX vendor wants to create is own HDLC driver, then it can create the Kconfig.xxx.

It's why the "xxx" extension of the Kconfig.xxx file is linked to the XXX vendor.

Reply:
The thing is you use `OPENTHREAD_HDLC_RCP` in `openthread.c`, which is vendor-agnostic implementation, we cannot introduce dependencies to vendor-speicifc configs in there.

So if the generic OT code needs some configs to change it's behavior when running in this host mode, we need some new Kconfig in the place I've mentioned above. A vendor-specific config here, enabling HDLC driver in this case, could select it.

Reply:
resolved